extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1588,6 +1617,83 @@ export class AmpA4A extends AMP.BaseElement {
     });
   }
 
+  /**
+   * @param {function()} checkStillCurrent
+   * @return {!Promise} Whether the creative was successfully rendered.
+   */
+  renderFriendlyTrustless_(checkStillCurrent) {
+    return this.sanitizedHeadPromise_.then((headData) => {
+      checkStillCurrent();
+      devAssert(this.element.ownerDocument);
+      this.maybeTriggerAnalyticsEvent_('renderFriendlyStart');
+
+      const {height, width} = this.creativeSize_;
+      const {extensions, fonts, head} = headData;
+      this.iframe = createSecureFrame(
+        this.element.ownerDocument,
+        head,
+        this.getIframeTitle(),
+        height,
+        width
+      );
+      this.applyFillContent(this.iframe);
```

Comment:
seems like quite a lot code duplication below this line. can we consolidate?

Reply:
Tried to remove duplication.